code/__HELPERS/icons.dm

```diff
@@ -1471,10 +1471,19 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
 /// If you want a stack trace to be output when the given state/file doesn't exist, use
 /// `/proc/icon_exists_or_scream()`.
 /proc/icon_exists(file, state)
-	var/static/list/icon_states_cache = list()
+	// monkestation start: icon_exists cache
+	var/static/list/icon_states_cache
+	if(isnull(icon_states_cache))
+		icon_states_cache = load_icon_exists_cache() || list()
+	// monkestation end
 	if(isnull(file) || isnull(state))
 		return FALSE //This is common enough that it shouldn't panic, imo.
 
+	// monkestation start: icon_exists cache
+	if(isfile(file) && !isnull(icon_states_cache["[file]"]?[state]))
+		return TRUE
+	// monkestation end
+
```

Comment:
This is already checked below. If the file doesn't exist, its `icon_states_cache` entry will be made into an empty list - and then the final return will return `FALSE`.

Reply:
this checks the actual *file path*

Reply:
Then could I recommend replacing this `if` statement and `return` statement with `file = "[file]"`? It would have the same result, without needing to be a file op or duplicated checks.